hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMDirectoryCreateRequest.java

```diff
@@ -160,52 +178,86 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       } else if (omDirectoryResult == DIRECTORY_EXISTS_IN_GIVENPATH ||
           omDirectoryResult == NONE) {
         dirKeyInfo = createDirectoryKeyInfo(ozoneManager, omBucketInfo,
-            volumeName, bucketName, keyName, keyArgs, transactionLogIndex);
+            volumeName, bucketName, keyName, keyArgs, trxnLogIndex);
 
         omMetadataManager.getKeyTable().addCacheEntry(
             new CacheKey<>(omMetadataManager.getOzoneKey(volumeName, bucketName,
                 dirKeyInfo.getKeyName())),
-            new CacheValue<>(Optional.of(dirKeyInfo), transactionLogIndex));
+            new CacheValue<>(Optional.of(dirKeyInfo), trxnLogIndex));
+
+        omClientResponse = new OMDirectoryCreateResponse(omResponse.build(),
+            dirKeyInfo);
+      } else {
+        // omDirectoryResult == DIRECTORY_EXITS
+        // Check if this is a replay of ratis logs
+        String dirKey = omMetadataManager.getOzoneDirKey(volumeName,
+            bucketName, keyName);
+        OmKeyInfo dbKeyInfo = omMetadataManager.getKeyTable().get(dirKey);
+        if (isReplay(ozoneManager, dbKeyInfo.getUpdateID(), trxnLogIndex)) {
+          throw new OMReplayException();
+        } else {
+          result = Result.DIRECTORY_ALREADY_EXISTS;
+          throw new OMException("Directory " + dirKey + "already exists in " +
+              "volume/bucket: " + volumeName + "/" + bucketName,
+              DIRECTORY_ALREADY_EXISTS);
+        }
       }
-      // if directory already exists do nothing or do we need to throw
-      // exception? Current KeyManagerImpl code does just return, following
-      // similar approach.
-
-      omResponse.setCreateDirectoryResponse(
-          CreateDirectoryResponse.newBuilder());
-      omClientResponse = new OMDirectoryCreateResponse(dirKeyInfo,
-          omResponse.build());
 
+      result = Result.SUCCESS;
     } catch (IOException ex) {
-      exception = ex;
-      omClientResponse = new OMDirectoryCreateResponse(null,
-          createErrorOMResponse(omResponse, exception));
+      if (ex instanceof OMReplayException) {
+        result = Result.REPLAY;
+        omClientResponse = new OMDirectoryCreateResponse(
+            createReplayOMResponse(omResponse));
+      } else {
+        if (result == null) {
```

Comment:
Minor NIT:
we can set result = Result.Failure in the beginning, in that way we don't need if check.